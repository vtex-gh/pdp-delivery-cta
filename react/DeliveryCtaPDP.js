import DeliveryCtaPDP from './components/DeliveryCtaPDP/index'

export default DeliveryCtaPDP
