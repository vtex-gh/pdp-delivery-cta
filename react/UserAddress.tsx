import React, { useEffect } from 'react'
import { useQuery } from 'react-apollo'
import { useCssHandles } from 'vtex.css-handles'
import type { CssHandlesTypes } from 'vtex.css-handles'
import ADDRESS_QUERY from 'vtex.store-resources/QueryAddress'

import Container from './Container'
import AddressInfo, {
  CSS_HANDLES as AddressInfoCssHandles,
} from './components/UserAddress/AddressInfo'
import { UserAddressCssHandlesProvider } from './components/UserAddress/UserAddressCssHandles'

export const USER_ADDRESS_CSS_HANDLES = [
  'userAddressContainer',
  ...AddressInfoCssHandles,
] as const

type Props = {
  /** Define if address should be displayed in a single line or in blocks */
  variation: 'inline' | 'bar'
  /** Define if street should be displayed */
  showStreet?: boolean
  /** Define if city and state should be displayed */
  showCityAndState?: boolean
  /** Define if postal code should be displayed */
  showPostalCode?: boolean
  /** Define if prefix should be displayed */
  showPrefix?: boolean
  /** Define if component should render if address is empty */
  showIfEmpty?: boolean
  /** Used to override default CSS handles */
  classes?: CssHandlesTypes.CustomClasses<typeof USER_ADDRESS_CSS_HANDLES>
}

function UserAddress({
  variation,
  showStreet = true,
  showCityAndState = false,
  showPostalCode = false,
  showPrefix = true,
  showIfEmpty = false,
  classes,
}: Props) {
  const { handles, withModifiers } = useCssHandles(USER_ADDRESS_CSS_HANDLES, {
    classes,
  })

  // todo: type the data returned
  const { data: orderFormData, refetch } = useQuery(ADDRESS_QUERY, {
    ssr: false,
  })

  useEffect(() => {
    const handleLocationUpdated = () => refetch()

    window.addEventListener('locationUpdated', handleLocationUpdated)

    return () => {
      window.removeEventListener('locationUpdated', handleLocationUpdated)
    }
  }, [refetch])

  const orderForm = orderFormData?.orderForm

  if (!orderForm || (!orderForm?.shippingData?.address && !showIfEmpty)) {
    return null
  }

  const isInline = variation === 'inline'
  const isInverted = !isInline

  const AddressInfoWithHandles = () => {
    return (
      <UserAddressCssHandlesProvider
        handles={handles}
        withModifiers={withModifiers}
      >
        <AddressInfo
          inverted={isInverted}
          inline={isInline}
          orderForm={orderForm}
          showStreet={showStreet}
          showCityAndState={showCityAndState}
          showPostalCode={showPostalCode}
          showPrefix={showPrefix}
          showIfEmpty={showIfEmpty}
        />
      </UserAddressCssHandlesProvider>
    )
  }

  if (isInline) {
    return (
      <div
        className={`${handles.userAddressContainer} ph5`}
        style={{
          maxWidth: '30rem',
        }}
      >
        <AddressInfoWithHandles />
      </div>
    )
  }

  return (
    <div
      className={`${handles.userAddressContainer} bg-base--inverted c-on-base--inverted flex ph5 pointer pv3 ml3 mr3`}
    >
      <Container className="flex justify-center w-100 left-0">
        <div className="w-100 mw9 flex">
          <AddressInfoWithHandles />
        </div>
      </Container>
    </div>
  )
}

export default UserAddress
