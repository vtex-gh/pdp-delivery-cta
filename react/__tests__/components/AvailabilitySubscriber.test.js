import React from 'react'
import { render } from '@vtex/test-tools/react'

import AvailabilitySubscriber from '../../AvailabilitySubscriber'

describe('<AvailabilitySubscriber />', () => {
  function renderComponent(customProps = {}) {
    const props = {
      skuId: '111111',
      available: false,
      ...customProps,
    }

    return render(<AvailabilitySubscriber {...props} />)
  }

  it('should be able to mount and not break', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })
})
