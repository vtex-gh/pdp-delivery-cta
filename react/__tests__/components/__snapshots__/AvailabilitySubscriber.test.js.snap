// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvailabilitySubscriber /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="subscriberContainer"
  >
    <div
      class="title t-body mb3"
    >
      This product isn't available right now
    </div>
    <div
      class="subscribeLabel t-small fw3"
    >
      I want to know when this product becomes available
    </div>
    <form
      class="form mb4"
    >
      <div
        class="content flex-ns justify-between mt4 mw6"
      >
        <div
          class="input inputName w-100 mr5 mb4"
        >
          <label>
            <input
              name="name"
              placeholder="Name"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="input inputEmail w-100 mr5 mb4"
        >
          <label>
            <input
              data-error="false"
              data-errormessage=""
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="submit flex items-center mb4"
        >
          <button
            data-isloading="false"
            data-variation="primary"
            disabled=""
            type="submit"
          >
            Send
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
